test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -676,9 +807,131 @@ class QueryTests {
     }
 
     @Test
-    @Ignore
+    fun sum_find_emptyTable() {
+        val assertions = { propertyDescriptor: PropertyDescriptor ->
+            realm.query(QuerySample::class)
+                .sum(propertyDescriptor.property.name, propertyDescriptor.clazz)
+                .find { sumValue: Any? ->
+                    if (sumValue is Number) {
+                        assertEquals(0, sumValue.toInt())
+                    }
+                }
+        }
+
+        // Iterate over non-nullable properties only - exclude RealmInstant
+        for (propertyDescriptor in propertyDescriptorsForSum) {
+            assertions(propertyDescriptor)
+        }
+
+        // Iterate over nullable properties containing both null and non-null values - exclude RealmInstant
+        for (nullablePropertyDescriptor in nullablePropertyDescriptorsForSum) {
```

Comment:
I would create an `allPropertyDescriptorsForSum` since that would make this an other tests a bit more readable IMO.